membership-attribute-service/app/controllers/AttributeController.scala

```diff
@@ -115,24 +114,34 @@ class AttributeController extends Controller with LoggingWithLogstashFields {
     def queryToAccountIds(response: QueryResponse): List[AccountId] =  response.records.map(_.Id)
 
     def getSubscriptions(accountIds: List[AccountId]): Future[Disjunction[String, List[Subscription[AnyPlan]]]] = {
-      def sub(accountId: AccountId): Future[Disjunction[String, List[Subscription[AnyPlan]]]] =
-        subscriptionService.subscriptionsForAccountId[AnyPlan](accountId)(anyPlanReads)
-
-      val maybeSubs: Future[Disjunction[String, List[Subscription[AnyPlan]]]] = accountIds.traverseU(id => sub(id)).map(_.sequenceU.map(_.flatten))
-
-      maybeSubs.map {
-        _.leftMap { errorMsg =>
-          log.error(s"We tried getting a subscription for a user with identityId ${identityId}, but then ${errorMsg}")
-          errorMsg
-        } map { subs =>
-          log.info(s"We got subs for identityId ${identityId} from Zuora and there were ${subs.length}")
-          subs
+      if(accountIds.isEmpty) Future.successful(\/.left(s"Can't get subscriptions for $identityId because they have no accountIds"))
+      else {
+        def sub(accountId: AccountId): Future[Disjunction[String, List[Subscription[AnyPlan]]]] =
+          subscriptionService.subscriptionsForAccountId[AnyPlan](accountId)(anyPlanReads)
+
+        val maybeSubs: Future[Disjunction[String, List[Subscription[AnyPlan]]]] = accountIds.traverseU(id => sub(id)).map(_.sequenceU.map(_.flatten))
+
+        maybeSubs.map {
+          _.leftMap { errorMsg =>
+            log.error(s"We tried getting a subscription for a user with identityId ${identityId}, but then ${errorMsg}")
+            errorMsg
+          } map { subs =>
+            log.info(s"We got subs for identityId ${identityId} from Zuora and there were ${subs.length}")
+            subs
+          }
         }
       }
     }
 
-    val attributes: DisjunctionT[Future, String, Option[Attributes]] = for {
-      accounts <- EitherT(withTimer(s"ZuoraAccountIdsFromIdentityId", zuoraRestService.getAccounts(identityId)))
+    def zuoraAccountsQuery(identityId: String): Future[Disjunction[String, QueryResponse]] = zuoraRestService.getAccounts(identityId).map {
+      _.leftMap { error =>
+        log.error(s"Calling ZuoraAccountIdsFromIdentityId failed for identityId ${identityId}. Error: ${error}")
+        error
+      }
+    }
+
+    val attributes = for {
+      accounts <- EitherT[Future, String, QueryResponse](withTimer(s"ZuoraAccountIdsFromIdentityId", zuoraAccountsQuery(identityId)))
       accountIds = queryToAccountIds(accounts)
       subscriptions <- EitherT[Future, String, List[Subscription[AnyPlan]]](withTimer(s"ZuoraGetSubscriptions", getSubscriptions(accountIds)))
```

Comment:
could have done 
```scala
subscriptions <- EitherT[Future, String, List[Subscription[AnyPlan]]](
  if (accountIds.nonEmpty) Future.successful(\/.right(Nil))
  else withTimer(s"ZuoraGetSubscriptions", getSubscriptions(accountIds)))
```

also it won't let me comment on the code, but ideally on 151 we could log the error
```scala
attributes.run.map(_.toOption).map(_.getOrElse(None))
```
could be
```scala
attributes.run.leftMap(err => logger.error(s"failed with: $error")).fold(_ => None, _)
```